src/fillalgebra.jl

```diff
@@ -104,27 +104,154 @@ function *(a::AbstractFillMatrix, b::Diagonal)
     a .* permutedims(b.diag) # use special broadcast
 end
 
-*(a::Adjoint{T, <:StridedMatrix{T}},   b::FillVector{T}) where T = reshape(sum(conj.(parent(a)); dims=1) .* b.value, size(parent(a), 2))
-*(a::Transpose{T, <:StridedMatrix{T}}, b::FillVector{T}) where T = reshape(sum(parent(a); dims=1) .* b.value, size(parent(a), 2))
-*(a::StridedMatrix{T}, b::FillVector{T}) where T         = reshape(sum(a; dims=2) .* b.value, size(a, 1))
+@noinline function check_matmul_sizes(y::AbstractVector, A::AbstractMatrix, x::AbstractVector)
+    Base.require_one_based_indexing(A, x, y)
+    size(A,2) == size(x,1) ||
+        throw(DimensionMismatch("second dimension of A, $(size(A,2)) does not match length of x $(length(x))"))
+    size(y,1) == size(A,1) ||
+        throw(DimensionMismatch("first dimension of A, $(size(A,1)) does not match length of y $(length(y))"))
+end
+@noinline function check_matmul_sizes(C::AbstractMatrix, A::AbstractMatrix, B::AbstractMatrix)
+    Base.require_one_based_indexing(A, B, C)
+    size(A,2) == size(B,1) ||
+        throw(DimensionMismatch("second dimension of A, $(size(A,2)) does not match first dimension of B, $(size(B,1))"))
+    size(C,1) == size(A,1) && size(C,2) == size(B,2) ||
+        throw(DimensionMismatch("A has size $(size(A)), B has size $(size(B)), C has size $(size(C))"))
+end
+
+function mul!(y::StridedVector, A::AbstractFillMatrix, b::AbstractFillVector, alpha::Number, beta::Number)
```

Comment:
Why is the restriction to `StridedVector` needed here?

Reply:
I've updated these now to accept more general destination arrays